adapters/repos/db/index.go

```diff
@@ -858,7 +858,9 @@ func (i *Index) objectSearch(ctx context.Context, limit int, filters *filters.Lo
 						delete(oo.Object.Additional, k)
 					}
 				}
-				oo.Object.Additional["explainScore"] = explainScore
+				if len(explainScore) > 0 {
```

Comment:
You already have additional.ExplainScore here, so maybe you could check that earlier (e.g. on 853) and also skip the entire loop?